scripts/generate_psa_key_attributes.py

```diff
@@ -0,0 +1,273 @@
+
+        self.key_type = key_type
+        self.lifetime = location
+        self.usage = usage
+        self.alg0 = algorithm
+        self.alg1 = PsaAlgorithm.NONE
+        self.bits = size
+        self.identifier = identifier
+
+        if location == PsaKeyLifetime.PERSISTENT_CRACEN_KMU:
+            if cracen_usage == PsaCracenUsageSceme.NONE:
+                print("--cracen_usage must be set if location target is PERSISTENT_CRACEN_KMU")
+                return
+            self.identifier = 0x7fff0000 | (cracen_usage << 12) | (identifier & 0xff)
+
+        if self.key_type == PsaKeyType.AES:
+            self.alg1 = PsaAlgorithm.NONE
+        elif self.key_type == PsaKeyType.ECC_TWISTED_EDWARDS:
+            self.alg1 = PsaAlgorithm.NONE
+        elif self.key_type == PsaKeyType.RAW_DATA:
+            self.alg1 = PsaAlgorithm.NONE
+        else:
+            raise RuntimeError("Invalid key type")
+
+    def pack(self):
+        """Builds a binary blob compatible with the psa_key_attributes_s C struct"""
+
+        return struct.pack(
+            "<hhIIIIII",
+            self.key_type,
+            self.bits,
+            self.lifetime,
+            # Policy
+            self.usage,
+            self.alg0,
+            self.alg1,
+            self.identifier,
+            0,  # Reserved, only used if key id encodes owner id
+        )
+
+def is_valid_hexa_code(string):
+    try:
+        int(string, 16)
+        return True
+    except ValueError:
+        return False
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        description="Generate PSA key attributes and write to stdout or"
+                    "create or append the information including the key to a"
+                    "nrfutil compatible json file. Also supports reading key"
+                    "from a PEM file in some cases",
+        allow_abbrev=False,
+    )
+
+    parser.add_argument(
+        "--usage",
+        help="Key usage",
+        type=str,
+        required=True,
+        choices=[x.name for x in PsaUsage],
+    )
+
+    parser.add_argument(
+        "--id",
+        help="Key identifier",
+        type=lambda number_string: int(number_string, 0),
+        required=True,
+    )
+
+    parser.add_argument(
+        "--type",
+        help="Key type",
+        type=str,
+        required=True,
+        choices=[x.name for x in PsaKeyType],
+    )
+
+    parser.add_argument(
+        "--size",
+        help="Key size in bits",
+        type=lambda number_string: int(number_string, 0),
+        required=True,
+    )
+
+    parser.add_argument(
+        "--algorithm",
+        help="Key algorithm",
+        type=str,
+        required=False,
+        default="NONE",
+        choices=[x.name for x in PsaAlgorithm],
+    )
+
+    parser.add_argument(
+        "--location",
+        help="Storage location",
+        type=str,
+        required=True,
+        choices=[x.name for x in PsaKeyLifetime],
+    )
+
+    parser.add_argument(
```

Comment:
So this is mutually exclusive with `--key-from-file`, but that is handled in the logic later in the script, not expressed in the parser itself? IMO it should be more clear that its either or. AFAICS its not mentioned in the docs either.